### DISCUSSION TITLE
Welcome to MDF Discussions!

### DISCUSSION THREAD

**Discussion by pgleeson** [ModECI/MDF#200] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Amna-28** [comment]
Hi!
Amna Ali here, I am an outreachy applicant for the upcoming batch and would like to start contributing to this project. 
Hope to work and learn together.

**Comment by Jingzcode** [comment]
Hi All,
I am Jing, graduated from KU Leuven last year majoring in Statistics and Data Science. I hope to start my open source journey via this project and look forward to contributing to it. 
Best regards,
Jing

**Comment by 0ddoes** [comment]
Hello all! My name is Izza. I am a Complex Systems and Dynamics MTech student from India. Really looking forward to being able to make meaningful contributions to this project. Can you please guide me on where we can begin or what to look into first?

**Comment by Srujana2505** [comment]
Hello everyone!!! I'm Srujana, I'm currently an undergraduate pursuing my bachelor's in Information Technology. I am interested in ML and I want to learn more by contributing to this project. Hope to work together and move forward.

**Comment by parikshit14** [comment]
Hi Team,

I am Parikshit Singh Rathore, I am a senior year undergraduate studying computer science and engineering. I have been contributing to open-source for a year and a half and would love to contribute & be part of this team for this summer through outreachy. 

Would be delighted to discuss project topics further and start working on them. 
Thanks and regards

**Comment by Mercytopsy** [comment]
Hello everyone
I'm Mercy who finished with computer Engineering degree. I'm one of the Outreachy Applicant for the Next Batch. I have knowledge of Data Science & ML, it will really be nice to be part of the team to contribute to this Open Source.

Thanks

**Comment by vishvarana** [comment]
Hello everyone,
I am Vishva, a final year student of computer engineering. I have been contributing to open source since almost an year and have loved every bit of it. It would be great opportunity to contribute here.

**Comment by FatimaArshad-DS** [comment]
Hi. My name is Fatima Arshad from Pakistan. I graduated from NUST as a Software Engineer and did my Masters in CS from LUMS, Pakistan. Hoping to be useful for this project.

**Comment by ghost** [comment]
Hi everyone, I am Nmeso an engineering graduate with an interest in ML. I would love to contribute to this project as I am certain that it would be really good for my self-development and would help me gain experience in open source.

**Comment by arpitadash** [comment]
Hello! My name is Arpita. I am an outreachy applicant. I am currently pursuing my Master's in computational neuroscience. I would like to contribute to this community and help improve the experience of sharing computational models across various applications and tools. I am also hoping to learn a lot in the process!

**Comment by salma2vec** [comment]
Hello everyone, 
I'm an applied machine learning engineer intern and computer science undegrad from India. I have been programming in Python for 2 years, and have fair experience working with Jupyter notebooks and semantic web. I have developed a keen interest in computational neuroscience research and I aim to gain experience in open source software development while contributing to this project, through Outreachy.

I've read the documentation, installed MDF & the required packages locally, and familiarized myself with the creation & execution of MDF models. Looking forward to learning a lot through contributing!

Regards, 
Salma

**Comment by Fifi-Banjoko** [comment]
Hello Everyone,

I am Fisayo, a Machine Learning Developer with a little over a year's experience with building machine learning models with Python. This is my first open source project and I am excited to learn and work with the team.

**Comment by chaitri19** [comment]
Hello Everyone,
I am Chaitri a 2nd year Mathematics and Computation student. I have been programming since a year and have some experience with Jupyter notebooks. I am very new to open source and this is my first open source project. I am very excited to join the team and learn with them.

**Comment by pradumna-gautam** [comment]
Hello Everone, I'm an Outreachy Applicant, looking forward to contributing to the organization.

**Comment by victorabba** [comment]
Hi My name is Victor. I'm an Outreachy applicant.  I would love to contribute to this project. I look forward to working with you all.

**Comment by sakshikaushik717** [comment]
Hi, my name is Sakshi Kaushik. I am excited to be a part of this community. I will get started with the Outreachy contribution guidelines and ask in case of doubts.

**Comment by somyagr** [comment]
Hi everyone! I am Somya Agrawal, pursuing Integrated Master of Technology in Mathematics and Computing. I am deeply enthusiastic about Machine learning and deep learning. I would love to contribute to this project, add more and more models to the MDF library and gain some valuable experience.

**Comment by sanjayankur31** [comment]
Welcome everyone (sorry, couldn't tag you all!). Please take a look at #213 for initial Outreachy tasks, and please open new topic whenver you need to. We're delighted to have so many of you interested in the project! :rocket:

**Comment by salma2vec** [comment]
Hi,
I've installed the MDF package and all dependencies, read the specification elements' documentation, ran the core MDF examples locally, and the notebook example using Jupyter notebook.
I'm in search of newcomer-friendly issues to work on and have commented on one, waiting to be assigned the particular task. Can I open pull requests without being assigned issues?
I'd like to know where I should go from here. Guidance would be appreciated. @pgleeson @sanjayankur31 

Looking forward to contributing to this project! 

Regards, 
Salma
<img width="706" alt="MDF" src="https://user-images.githubusercontent.com/65449934/160674160-62de7b21-d694-4754-9bb4-93fcaad9c415.png">

**Reply by pgleeson** [reply]
Thanks @IdealisticINTJ. Yes, please feel free to open PRs if you think you have something you'd like to contribute. We'll work to add some more entry level issues over the next few days.

**Reply by salma2vec** [reply]
Got it. Thanks!

**Comment by shivani6320** [comment]
Hello Everyone!! I'm Shivani - an outreachy applicant🤓. I am super excited to be part of this project and learn from the mentors and fellow community members. Looking forward to making meaningful contributions.✨

**Comment by Fifi-Banjoko** [comment]
Hello Everyone,

In case anyone needs a little guidance to get started especially if you are new to using Git and Github, here are the steps I took to install the MDF package and dependencies and also to run locally the MDF examples. It should not take too long and it's done from the command line interface (CLI).

P.S. I am not an expert and there may be better and faster ways to do this :).

1.) If you do not have the Git software you'll have to download it since you cannot use Github without Git. You can find it online. When that is done, if you input "git" it into the CLI it should return the Git version, commands and such.
2.) Use "git clone https://github.com/ModECI/MDF.git" to clone the MDF repository (which has all the project files, source code and versions) to your computer so you can have a local repository.
3.) Change the directory to the MDF directory using the 'cd' command followed by the MDF file path.
4.) Create a virtual environment in the MDF directory. If you don't have virtualenv installed you'll need to install it. You may use "pip install virtualenv".
5.) Activate the virtual environment. 
6.) Install the packages from the repository. You may use "python setup.py install" and "pip install ."
7.) You can create a branch using "git branch *name of branch*". If you run "git branch" it should return the branch name you created.
8.) Your current directory should be the main MDF directory. Change it to the directory of the example you would like to run and run the example(s).

Let me know if you have question or if I missed something or got something wrong.

**Comment by KhadijaKamran** [comment]
Hey everyone! I am a newcomer and I want to contribute to open source. I have read the initial contribution guidelines for MDF. A quick question, is there a specific python version recommended? Kindly let me no so that I start working. Thank you! :)

**Reply by sanjayankur31** [reply]
Hi @KhadijaKamran , welcome!

All currently supported python version should work, but I think there are a few dependencies that haven't yet included py311 support, so that one may be a little flaky. See:

https://github.com/ModECI/MDF/issues/362

**Reply by KhadijaKamran** [reply]
Okay, I will install python 3.9 then. Thank you for the reply. :)

**Reply by KhadijaKamran** [reply]
Hey @sanjayankur31! I have created my venv and I am using Windows 10. I have tried a lot to activate my venv from cmd. Scripts\activate.bat from venv dir is not working. :(
I am also trying to activate it on VS Code but nothing seems to work. Can you please guide me or recommend me some readings that can help me figure this out. 😇

**Reply by KhadijaKamran** [reply]
I ran my Power Shell as an Administrator and run "Set-ExecutionPolicy RemoteSigned". Then I moved to my MDF repository where my virtual environment was created. Running "venv_name\Scripts\activate" seems to work now. 😄  
By default execution policy is 'Restricted' and it does not run scripts file. Changing it to 'RemoteSigned' allows to run scripts that are signed by a trusted publisher. Thank you for taking the time to read my message :)

**Reply by sanjayankur31** [reply]
I'm glad it worked. I haven't used Windows in > 10 years, unfortunately, so I'm not very well versed with how things work there. We've got instructions in the docs here that have been verified by others:

https://mdf.readthedocs.io/en/development/api/Installation.html#id1

**Comment by ObianujuCarol** [comment]
Hi everyone, 
My name is Obianujunwa Nwaiwu, I am so excited to be here and as well this is my first open-source contribution. I hope to make meaningful contributions to the success of the projects and learn on the process. 
Thank you all.

**Reply by sanjayankur31** [reply]
Welcome @ObianujuCarol , very happy to have you here :)

**Comment by Aeeshah1** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.my 
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Hello, my name is Aishat Ijiyode, I'm an outreachy applicant for the May 2024 cohort

**Comment by Iqra350** [comment]
Hi! My name is Iqra. I also have run the repo and sample jupyter notebooks. looking forward to more contribution to that project. In order to make it easy to understand can you give us some rough examples for contributions? Any hint to go to right direction or what you are expecting from us.

**Comment by Talikamuhib** [comment]
Hello World! 👍 

Taliqa here from Pakistan - A crazy curly short hair girl with light brown eyes. I am an Outreachy Contributor!

Happy Contributing : )

**Comment by pgleeson** [comment]
Hi all, see here: https://github.com/ModECI/MDF/discussions/509 for advice on the May 2024 project.

**Comment by AshitaSri** [comment]
Hello everyone, I am Ashita Srivastava, this will me first contribution. I looking forward to great learning.

**Comment by riddhihalade** [comment]
Hello everyone,

I'm Riddhi Halade, an Outreachy applicant and aspiring software engineer. I'm incredibly excited to learn and collaborate with this Project community. While I'm eager to contribute, I'm also new to the project and would be grateful for any guidance on the best first steps I can take to effectively familiarize myself with MDF and its goals.

**Comment by pgleeson** [comment]
Hi @riddhihalade, see here: https://github.com/ModECI/MDF/discussions/509 for advice on the May 2024 project.

**Comment by sandeepB3** [comment]
Hey @pgleeson, in `examples/MDF/simple.py`, shouldn't os be imported? As in line 94 and line 104, it shows the warning: 
`"os" is not defined`, and in other examples like abcd.py and arrays.py the `import os` line is defined. Although even without importing os, the code in `simple.py` runs just fine.

**Reply by sanjayankur31** [reply]
Yes, it should be imported. Could you open a pull request against the development branch please?

If you run it with the `-graph` argument, the script will error---the `os.name` line is only executed when this is used.

**Reply by sandeepB3** [reply]
Sure will do that, thanks!

**Reply by sandeepB3** [reply]
PR #510

**Comment by sanjayankur31** [comment]
Everyone, please do look at #509 and use that for discussions. It just helps us keep it all in one place :)

**Comment by Talikamuhib** [comment]
HELLO WORLD! TALIKA HERE

**Comment by ektuu** [comment]
Hi everyone!
I’m Ekta and I’m really excited to get involved in open source communities and projects.
I’m a second year undergrad from India pursuing Computer science as my majors.

I’ve been indulged in Machine learning since last year, and I came across this project  and it really aligns with my interests.

Hope to have lots of learning!

**Reply by rasmodev** [reply]
@ektasiwach Welcome to the community. To begin with the contributions, please go to [https://github.com/ModECI/MDF/discussions/509](url) for the first 3 tasks to catch up with everyone else.

Happy contribution stage. Let's connect more.

**Reply by ektuu** [reply]
@rasmodev Thanks for pointing me in the right direction! I've already checked out the link at https://github.com/ModECI/MDF/discussions/509 and am working on the tasks. Excited to contribute and engage with the community further!

**Comment by codinghappiness-web** [comment]
Hi everyone
I am Happiness
I am a Data Scientist and Technical Writer
I am happy to be contributing to this project
